src/app/loadout-builder/filter/LoadoutOptimizerMenuItems.tsx

```diff
@@ -230,61 +95,221 @@ export default memo(function LockArmorAndPerks({
+   */
+  const lockEquipped = () =>
+    lbDispatch({
+      type: 'setPinnedItems',
+      items: selectedStore.items.filter((item) => item.equipped && isLoadoutBuilderItem(item)),
+    });
+
+  const pinItem = useCallback(
+    (item: DimItem) => lbDispatch({ type: 'pinItem', item }),
+    [lbDispatch]
+  );
+  const unpinItem = (item: DimItem) => lbDispatch({ type: 'unpinItem', item });
+
+  const chooseLockItem = chooseItem(
+    pinItem,
+    // Exclude types that already have a locked item represented
+    (item) => Boolean(!pinnedItems[item.bucket.hash] && searchFilter(item))
+  );
+
+  const allPinnedItems = _.sortBy(_.compact(objectValues(pinnedItems)), (i) =>
+    LockableBucketHashes.indexOf(i.bucket.hash)
+  );
+
+  return (
+    <LoadoutBucketDropTarget className={styles.area} onItemLocked={pinItem}>
+      {Boolean(allPinnedItems.length) && (
+        <div className={styles.itemGrid}>
+          {allPinnedItems.map((lockedItem) => (
+            <LockedItem key={lockedItem.id} lockedItem={lockedItem} onRemove={unpinItem} />
+          ))}
+        </div>
+      )}
+      <div className={styles.buttons}>
+        <button type="button" className="dim-button" onClick={chooseLockItem}>
+          <AppIcon icon={pinIcon} /> {t('LoadoutBuilder.LockItem')}
+        </button>
+        <button type="button" className="dim-button" onClick={lockEquipped}>
+          <AppIcon icon={pinIcon} /> {t('LoadoutBuilder.LockEquipped')}
+        </button>
+      </div>
+    </LoadoutBucketDropTarget>
+  );
+});
+
+export const LoadoutOptimizerExcludedItems = memo(function LoadoutOptimizerExcludedItems({
+  chooseItem,
+  excludedItems,
+  searchFilter,
+  lbDispatch,
+}: {
+  chooseItem: ChooseItemFunction;
+  excludedItems: ExcludedItems;
+  searchFilter: ItemFilter;
+  lbDispatch: Dispatch<LoadoutBuilderAction>;
+}) {
+  const excludeItem = useCallback(
+    (item: DimItem) => lbDispatch({ type: 'excludeItem', item }),
+    [lbDispatch]
+  );
+  const unExcludeItem = (item: DimItem) => lbDispatch({ type: 'unexcludeItem', item });
+
+  const chooseExcludeItem = chooseItem(excludeItem, (item) => Boolean(searchFilter(item)));
+
+  const allExcludedItems = _.sortBy(_.compact(objectValues(excludedItems)).flat(), (i) =>
+    LockableBucketHashes.indexOf(i.bucket.hash)
+  );
+  return (
+    <LoadoutBucketDropTarget className={styles.area} onItemLocked={excludeItem}>
+      {Boolean(allExcludedItems.length) && (
+        <div className={styles.itemGrid}>
+          {allExcludedItems.map((lockedItem) => (
+            <LockedItem key={lockedItem.id} lockedItem={lockedItem} onRemove={unExcludeItem} />
+          ))}
+        </div>
+      )}
+      <div className={styles.buttons}>
+        <button type="button" className="dim-button" onClick={chooseExcludeItem}>
+          <AppIcon icon={faTimesCircle} /> {t('LoadoutBuilder.ExcludeItem')}
+        </button>
+      </div>
+    </LoadoutBucketDropTarget>
+  );
+});
+
+export const loMenuSection = styles.area;
```

Comment:
Maybe a spicy opinion but I don't love this. I would personally either just import these styles elsewhere or if you are concerned about encapsulation stick this in another css module.

Reply:
IMO it's the least bad of the alternatives:

1. Import directly from the stylesheet to somewhere else - works, but then our eslint plugin that tells us when styles are not used will get confused. I don't have a better solution for identifying unused selectors right now.
2. Using `composes X from Y` in another CSS module - fine, but if you don't want to add anything you're emitting an empty CSS block and class name.
3. Use a wrapper component (e.g. `function Foo() { return <div className={styles.area}>{children}</div> }`). In this case that's awkward because I'm applying the area class to all kinds of different elements and I don't always want an extra wrapper div just for this.
4. Copy/paste the rule - this happens a lot
5. Re-export class names as part of the module's external interface - easy to consume, keeps the styles "with" the owning module.